server/src/dispatcher.erl

```diff
@@ -0,0 +1,145 @@
+    {reply, Reply, State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Handling cast messages
+%%
+%% @spec handle_cast(Msg, State) -> {noreply, State} |
+%%                                  {noreply, State, Timeout} |
+%%                                  {stop, Reason, State}
+%% @end
+%%--------------------------------------------------------------------
+handle_cast(_Msg, State) ->
+    {noreply, State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Handling all non call/cast messages
+%%
+%% @spec handle_info(Info, State) -> {noreply, State} |
+%%                                   {noreply, State, Timeout} |
+%%                                   {stop, Reason, State}
+%% @end
+%%--------------------------------------------------------------------
+handle_info({From, {login, Request}}, State) ->
+    io:format("[~p] INFO: ~p~n", [?SERVER, From]),
+    login(State, From, Request),
+    {noreply, State};
+handle_info(Info, State) ->
+    io:format("[~p] INFO: ~p~n", [?SERVER, Info]),
+    {noreply, State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% This function is called by a gen_server when it is about to
+%% terminate. It should be the opposite of Module:init/1 and do any
+%% necessary cleaning up. When it returns, the gen_server terminates
+%% with Reason. The return value is ignored.
+%%
+%% @spec terminate(Reason, State) -> void()
+%% @end
+%%--------------------------------------------------------------------
+terminate(_Reason, _State) ->
+    ok.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Convert process state when code is changed
+%%
+%% @spec code_change(OldVsn, State, Extra) -> {ok, NewState}
+%% @end
+%%--------------------------------------------------------------------
+code_change(_OldVsn, State, _Extra) ->
+    {ok, State}.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+login(State, From, #{username := Username, password := _Password} = Request) ->
+    io:format("[~p] User ~p is attempting to login from ~p~n", [?SERVER, Username, From]),
+    case registry:check_user(Request, State#state.connection) of
+        {ok, Email} ->
+            io:format("[~p] USER: ~p successfully logged in!~n", [?SERVER, Email]),
+            {ok, Connection} = database:database_connect(),
+            PlayerState = #state{pid = From, connection = Connection},
+            io:format("[~p] Setting USER_STATE=~p...~n", [?SERVER, PlayerState]),
+            {ok, Pid} = player_sup:start([PlayerState]),
+            io:format("[~p] Successfully Spawned ~p~n", [?SERVER, Pid]),
+            From ! {ok, {Pid, Email}};
+        {error, Message} ->
+            io:format("Unexpected Error: ~p~n", [Message]),
```

Comment:
```suggestion
            io:format("Failed to login: ~p~n", [Message]),
```
Let's use this opportunity to improve other minor details. I would like more specific logs in the server. What do you think? (we may apply this to all internal functions as well).